src/components/network/geo-data.js

```diff
@@ -81,11 +83,77 @@ export default class GeoData {
                     }
                     positions[positions.length - 1] = substationPosition1;
                 }
-
                 return positions;
             }
         }
 
         return [ substationPosition1, substationPosition2 ];
     };
+
+    getLineDistances(positions) {
+        if (positions !== null && positions.length > 1) {
+            let cumulativeDistanceArray = [0];
+            let cumulativeDistance = 0;
+            let segmentDistance;
+            let ruler;
+            for (let i = 0; i < positions.length - 1; i++) {
+                ruler = cheapRuler(positions[i][1], 'meters');
+                segmentDistance = ruler.lineDistance(positions.slice(i, i + 2));
+                cumulativeDistance = cumulativeDistance + segmentDistance;
+                cumulativeDistanceArray[i+1] = cumulativeDistance;
+            }
+            return cumulativeDistanceArray;
+        }
+        return null;
+    }
+
+    /**
+     * Find the polyline in which we reach the wanted distance and return the polyline
+     * along with the distance left to travel on this polyline to be at the exact wanted distance
+     * (implemented using a binary search)
+     */
+    findPolyline(positions, cumulativeDistances, wantedDistance) {
```

Comment:
Rename to findSegment (a polyline is typically more than 1 segment) ?
Also rename "leftDistance" to "remainingDistance" ?